src/test/kotlin/org/rust/cargo/runconfig/TestRunnerTest.kt

```diff
@@ -0,0 +1,370 @@
+                    mod test_mod {
+                        #[test]
+                        fn test() {}
+                    }
+
+                    #[test]
+                    fn test() {}
+                """)
+            }
+        }
+        myFixture.configureFromTempProjectFile(testProject.fileWithCaret)
+        val configuration = createTestRunConfigurationFromContext()
+        val root = runConfigurationAndGetTestRoot(configuration)
+
+        val test = findTestByName("tests::test", root)!!
+        assertEquals("cargo:test://tests::test", test.locationUrl)
+
+        val mod = findTestByName("tests::test_mod", root)!!
+        assertEquals("cargo:suite://tests::test_mod", mod.locationUrl)
+
+        val testInner = findTestByName("tests::test_mod::test", root)!!
+        assertEquals("cargo:test://tests::test_mod::test", testInner.locationUrl)
+    }
+
+    fun `test test duration`() {
+        val testProject = buildProject {
+            toml("Cargo.toml", """
+                [package]
+                name = "sandbox"
+                version = "0.1.0"
+                authors = []
+
+                [[test]]
+                name = "tests"
+                path = "tests/tests.rs"
+            """)
+
+            dir("tests") {
+                rust("tests.rs", """
+                    use std::thread;
+                    /*caret*/
+
+                    #[test]
+                    fn test1() {
+                        thread::sleep_ms(2000);
+                    }
+
+                    #[test]
+                    fn test2() {}
+                """)
+            }
+        }
+        myFixture.configureFromTempProjectFile(testProject.fileWithCaret)
+        val configuration = createTestRunConfigurationFromContext()
+        val root = runConfigurationAndGetTestRoot(configuration)
+
+        val test1 = findTestByName("tests::test1", root)!!
+        assert(test1.duration!! > 1000)
+
+        val test2 = findTestByName("tests::test2", root)!!
+        assert(test2.duration!! < 100)
+
+        val mod = findTestByName("tests", root)!!
+        assert(mod.duration!! == test1.duration!! + test2.duration!!)
+    }
+
+    private fun createTestRunConfigurationFromContext(): CargoCommandConfiguration {
+        val context = DataManager.getInstance().getDataContext(myFixture.editor.component)
+        return CargoTestRunConfigurationProducer()
+            .createConfigurationFromContext(ConfigurationContext.getFromContext(context))
+            ?.configuration as? CargoCommandConfiguration
+            ?: error("Can't create run configuration")
+    }
+
+    private fun runConfigurationAndGetTestRoot(configuration: RunConfiguration): SMTestProxy.SMRootTestProxy {
+        val executor = DefaultRunExecutor.getRunExecutorInstance()
+        val state = ExecutionEnvironmentBuilder
+            .create(executor, configuration)
+            .build()
+            .state!!
+        val result = state.execute(executor, RsRunner())!!
```

Comment:
The first part of this method is almost the same as in `org.rust.cargo.runconfig.RunConfigurationTestCase#execute`. Let's reuse it somehow